endogenous/scripts/end_transport.py

```diff
@@ -0,0 +1,186 @@
+#!/usr/bin/env python3
+# -*- coding: utf-8 -*-
+"""
+Created on Tue Feb 14 16:01:01 2023
+
+@author: au731993
+"""
+
+import pypsa 
+import pandas as pd
+import numpy as np
+
+#makes it possible to add more busses to Links
+override_component_attrs = pypsa.descriptors.Dict(
+    {k: v.copy() for k, v in pypsa.components.component_attrs.items()}
+)
+override_component_attrs["Link"].loc["bus2"] = [
+    "string",
+    np.nan,
+    np.nan,
+    "2nd bus",
+    "Input (optional)",
+]
+override_component_attrs["Link"].loc["bus3"] = [
+    "string",
+    np.nan,
+    np.nan,
+    "3rd bus",
+    "Input (optional)",
+]
+override_component_attrs["Link"].loc["efficiency2"] = [
+    "static or series",
+    "per unit",
+    1.0,
+    "2nd bus efficiency",
+    "Input (optional)",
+]
+override_component_attrs["Link"].loc["efficiency3"] = [
+    "static or series",
+    "per unit",
+    1.0,
+    "3rd bus efficiency",
+    "Input (optional)",
+]
+override_component_attrs["Link"].loc["p2"] = [
+    "series",
+    "MW",
+    0.0,
+    "2nd bus output",
+    "Output",
+]
+override_component_attrs["Link"].loc["p3"] = [
+    "series",
+    "MW",
+    0.0,
+    "3rd bus output",
+    "Output",
+]
+
+def annuity(n,r):
+    """ Calculate the annuity factor 
+    for an asset with lifetime n years and 
+    discount rate of f, e.g. annuity(20,0.05)*20 = 1.6 """
+    
+    if r > 0:
+        return r/(1. - 1./(1.+r)**n)
+    else:
+        return 1/n
+
+def transport(string):
+    network = pypsa.Network(override_component_attrs=override_component_attrs)
+    
+    hours_in_2013 = pd.date_range('2013-01-01T00:00Z','2013-12-31T23:00Z', freq='H')
+    network.set_snapshots(hours_in_2013)
+    
+    df_tech = pd.read_csv(string + 'resources/costs_2025.csv', sep=',', index_col=0)
+    df_oil = df_tech.loc['oil']
+    oil = df_oil.set_index(['parameter'])
+    #discount rate 0.07
+    oil_capital_cost = annuity(oil.at['lifetime','value'], 0.07)*oil.at['investment','value']*1000*(1+oil.at['FOM','value']/100) # in €/MW
+    oil_marginal_cost = oil.at['VOM','value']
+    
+    network.add("Carrier", "oil", co2_emissions=1.)
+    
+    network.add("Bus", 
+                "oil bus", 
+                carrier="oil")
+    
+    network.add("Generator",
+                'oil',
+                bus="oil bus",
+                p_nom=100,
```

Comment:
AFAIK, setting _p_nom=100_ is redundant/unecessary when _p_nom_extendable=True_ (i.e. _p_nom=100_ can safely be removed).